projects/ad713x_fmcz/ad713x_fmc.c

```diff
@@ -245,6 +273,80 @@ int main()
 	msg->tx_buf_addr = 0xA000000;
 	msg->rx_buf = calloc((AD7134_FMC_CH_NO * AD7134_FMC_SAMPLE_NO),
 			     sizeof(uint32_t));
+
+#ifdef USE_LIBIIO
+	struct iio_ad713x *iio_ad713x;
+	struct iio_server_ops uart_iio_server_ops;
+	struct iio_app_init_param iio_app_init_par;
+
+	struct xil_irq_init_param xil_irq_init_par = {
+		.type = IRQ_PS,
+	};
+
+	struct irq_init_param irq_init_param = {
+		.irq_ctrl_id = INTC_DEVICE_ID,
+		.extra = &xil_irq_init_par,
+	};
+
+	struct irq_ctrl_desc *irq_desc;
+
+	struct xil_uart_init_param xil_uart_init_par = {
+		.type = UART_PS,
+		.irq_id = UART_IRQ_ID,
+		.irq_desc = irq_desc,
+	};
+
+	struct iio_app_desc *iio_app_desc;
+
+	struct uart_init_param uart_init_par = {
+		.baud_rate = 115200,
+		.device_id = UART_DEVICE_ID,
+		.extra = &xil_uart_init_par,
+	};
+
+	char dev_name[] = "adc";
+	struct iio_ad713x_init_par iio_ad713x_init_par = {
+		.name = dev_name,
+		.num_channels = 8,
+		.spi_eng_desc = spi_eng_desc,
+		.msg = msg,
+		.dcache_invalidate_range = (void (*)(uint32_t, uint32_t))Xil_DCacheInvalidateRange,
+	};
+
+	ret = iio_ad713x_init(&iio_ad713x, &iio_ad713x_init_par);
+	if(ret < 0)
+		return ret;
+
+	uart_iio_server_ops = (struct iio_server_ops) {
+		.read = iio_uart_read,
```

Comment:
Shouldn't iio_server_ops have a kind of void* dev for read and write?

Reply:
This will mean changes into libtinyiiod library:
https://github.com/analogdevicesinc/libtinyiiod/blob/6d696c5bda78d8a07052dd6738d780e90a677f8b/tinyiiod.h#L27
Do we want that?

Reply:
Not in this pull request